torch/cuda/__init__.py

```diff
@@ -693,19 +702,55 @@ def utilization(device: Optional[Union[Device, int]] = None) -> int:
     Warning: Each sample period may be between 1 second and 1/6 second,
     depending on the product being queried.
     """
-    try:
-        import pynvml  # type: ignore[import]
-    except ModuleNotFoundError as e:
-        raise ModuleNotFoundError("pynvml module not found, please install pynvml") from e
-    from pynvml import NVMLError_DriverNotLoaded
-    try:
-        pynvml.nvmlInit()
-    except NVMLError_DriverNotLoaded as e:
-        raise RuntimeError("cuda driver can't be loaded, is cuda enabled?") from e
-    device = _get_device_index(device, optional=True)
-    handle = pynvml.nvmlDeviceGetHandleByIndex(device)
+    handle = _setup_pynvml(device)
     return pynvml.nvmlDeviceGetUtilizationRates(handle).gpu
 
+def temperature(device: Optional[Union[Device, int]] = None) -> int:
+    r"""Returns the average temperature of the GPU sensor over the past sample period as given by `nvidia-smi`.
+
+    Args:
+        device (torch.device or int, optional): selected device. Returns
+            statistic for the current device, given by :func:`~torch.cuda.current_device`,
+            if :attr:`device` is ``None`` (default).
+
+    Warning: Each sample period may be between 1 second and 1/6 second,
+    depending on the product being queried.
+    """
+    handle = _setup_pynvml(device)
+    # 0 refers to the temperature sensor for the GPU die.
+    return pynvml.nvmlDeviceGetTemperature(handle, 0)
+
+def power(device: Optional[Union[Device, int]] = None) -> int:
+    r"""Returns the average power draw of the GPU sensor over the past sample period as given by `nvidia-smi`.
+
+    Args:
+        device (torch.device or int, optional): selected device. Returns
+            statistic for the current device, given by :func:`~torch.cuda.current_device`,
+            if :attr:`device` is ``None`` (default).
+
+    Warning: Each sample period may be between 1 second and 1/6 second,
+    depending on the product being queried.
+    """
+    handle = _setup_pynvml(device)
+    return pynvml.nvmlDeviceGetPowerUsage(handle)
+
+def clock_speed(device: Optional[Union[Device, int]] = None) -> int:
+    r"""Returns the clock speed of the GPU SM over the past sample period as given by `nvidia-smi`.
+
+    Args:
+        device (torch.device or int, optional): selected device. Returns
+            statistic for the current device, given by :func:`~torch.cuda.current_device`,
+            if :attr:`device` is ``None`` (default).
+
+    Warning: Each sample period may be between 1 second and 1/6 second,
+    depending on the product being queried.
+    """
+    handle = _setup_pynvml(device)
+    # Clocks documented here https://docs.nvidia.com/deploy/nvml-api/group__nvmlDeviceEnumvs.html#group__nvmlDeviceEnumvs_1g805c0647be9996589fc5e3f6ff680c64
```

Comment:
This is minor, feel free to ignore it.
I'm not a huge fan of external links in comments unless it's something that should be almost eternal (e.g. a PR number, an arxiv or doi). Though I think it's helpful to point to it here. Can we move this to the PR discussion for instance, and redirect this comment there? Like this if the link breaks and someone tries it we'll be notified by a message on the PR.